test/run-drun/ok/oom-query.drun-run.ok

```diff
@@ -1,3 +1,3 @@
 ingress Completed: Reply: 0x4449444c016c01b3c4b1f204680100010a00000000000000000101
 ingress Completed: Reply: 0x4449444c0000
-Err: IC0503: Canister rwlgt-iiaaa-aaaaa-aaaaa-cai trapped explicitly: RTS error: Cannot grow memory
+Err: IC0502: Canister rwlgt-iiaaa-aaaaa-aaaaa-cai trapped: heap or table out of bounds
```

Comment:
Wait, why is this message changing? That suggests the overflow detection isn't kicking in and we are trapping due to corruption.

Reply:
These tests don't run on CI, and for some reason I can't run them locally. @luc-blaeser could you run them and accept the new output?

Reply:
Now I get
``` diff
--- oom-query.drun-run (expected)
+++ oom-query.drun-run (actual)
@@ -1,3 +1,3 @@
 ingress Completed: Reply: 0x4449444c016c01b3c4b1f204680100010a00000000000000000101
 ingress Completed: Reply: 0x4449444c0000
-Err: IC0502: Canister rwlgt-iiaaa-aaaaa-aaaaa-cai trapped: heap or table out of bounds
+Err: IC0522: Canister rwlgt-iiaaa-aaaaa-aaaaa-cai exceeded the instruction limit for single message execution.
Some tests failed:
oom-query.mo
```

Reply:
Ok, I get that too now.
(on my PR branch)
Need to check master.

Reply:
Okay, I think I have a solution for this. I'll introduce a `prim` to advance the `HP` to a certain address, marking the heap segment as `TAG_FREE_SPACE`. This is super cheap, and gives us the tool to increase heap pressure. No need for expensive 3 GB arrays any more.
What do you think @luc-blaeser @crusso?
I'll do a separate PR for this.

Reply:
But that wouldn't permit precise adjustment of the `HP`, which is what the first two TODO items ask for :-(